domain/resource/service/resource.go

```diff
@@ -167,12 +190,34 @@ func (s *Service) GetApplicationResourceID(
 	if err := args.ApplicationID.Validate(); err != nil {
 		return "", errors.Errorf("application id: %w", err)
 	}
-	if args.Name == "" {
+	if strings.TrimSpace(args.Name) == "" {
 		return "", resourceerrors.ResourceNameNotValid
 	}
 	return s.st.GetApplicationResourceID(ctx, args)
 }
 
+// GetResourceUUIDByApplicationAndResourceName returns the ID of the application
+// resource specified by natural key of application and resource name.
+//
+// The following error types can be expected to be returned:
+//   - [resourceerrors.ApplicationNotFound] is returned if the application is
+//     not found.
+//   - [resourceerrors.ResourceNotFound] if no resource with name exists for
+//     given application.
+func (s *Service) GetResourceUUIDByApplicationAndResourceName(
+	ctx context.Context,
+	appName string,
+	resName string,
+) (coreresource.UUID, error) {
+	if strings.TrimSpace(resName) == "" {
```

Comment:
todo: Please use `if resName == "" {`. We shouldn't add a restriction that does not exist when the charm is unmarshalled.

Here is the code in internal/charm which validates a resource name:
https://github.com/juju/juju/blob/b7a80a1457d89742973e989ab5b67c74be166dd6/internal/charm/resource/meta.go#L38-L41